sdk/storage/Azure.Storage.Blobs.Batch/src/BlobBatch.cs

```diff
@@ -224,7 +224,7 @@ private void SetBatchOperationType(BlobBatchOperationType operationType)
                             BlobDeleteHeaders blobDeleteHeaders = new BlobDeleteHeaders(response);
                             return ResponseWithHeaders.FromValue(blobDeleteHeaders, response);
                         default:
-                            throw await _clientDiagnostics.CreateRequestFailedExceptionAsync(response).ConfigureAwait(false);
+                            throw new RequestFailedException(response);
```

Comment:
Here and below, because you removed the only `await` the delegate should not be `async`.